Assets/Scripts/Menus/DisplayManager.cs

```diff
@@ -0,0 +1,278 @@
+    }
+
+    /// <summary>
+    /// Applies the selected display mode
+    /// </summary>
+    /// <param name="index">Display mode index</param>
+    private void ApplyDisplayMode(int index)
+    {
+        switch (index)
+        {
+            case 0: // Borderless
+                Screen.fullScreen = true;
+                break;
+            case 1: // Windowed
+                Screen.fullScreen = false;
+                break;
+            default:
+                Debug.LogError("Invalid display mode selected!");
+                break;
+        }
+    }
+    
+    /// <summary>
+    /// Applies the selected resolution
+    /// </summary>
+    /// <param name="index">Resolution index</param>
+    private void ApplyResolution(int index)
+    {
+        Resolution selectedResolution = resolutions[index];
+        Screen.SetResolution(selectedResolution.width, selectedResolution.height, Screen.fullScreen);
+    }
+
+    /// <summary>
+    /// Applies the selected frame rate
+    /// </summary>
+    /// <param name="index">Frame rate index</param>
+    private void ApplyFrameRate(int index)
+    {
+        string selectedOption = frameRateDropdown.options[index].text;
+        
+        switch (selectedOption)
+        {
+            case "30 FPS":
+                Application.targetFrameRate = 30;
+                break;
+            case "60 FPS":
+                Application.targetFrameRate = 60;
+                break;
+            case "90 FPS":
+                Application.targetFrameRate = 90;
+                break;
+            case "120 FPS":
+                Application.targetFrameRate = 120;
+                break;
+            case "144 FPS":
+                Application.targetFrameRate = 144;
+                break;
+            case "240 FPS":
+                Application.targetFrameRate = 240;
+                break;
+            case "Uncapped":
+                Application.targetFrameRate = -1; // Uncapped
+                break;
+            default:
+                Debug.LogError("Invalid frame rate option selected!");
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Applies the VSync setting
+    /// </summary>
+    private void ApplyVsync(bool value)
+    {
+        if (value) //VSyncCount expects a 0 or a 1 not true or false so I have to convert it here
+        {
+            QualitySettings.vSyncCount = 1;
+        }
+        else
+        {
+            QualitySettings.vSyncCount = 0;
+        }
+    }
+
+    /// <summary>
+    /// Loads the settings from PlayerPrefs
+    /// </summary> 
+    private void LoadSettings()
+    {
+        //To test default values uncomment the line below
+        //PlayerPrefs.DeleteAll();
+
+        //These are all inside try catch blocks as the PlayerPrefs values could have index out of range errors if the player changes monitors to one that doesn't support the resolution they had set
```

Comment:
I was about to add that the resolution settings save, but none of the other settings save.

Reply:
Resolved